events/equeue/equeue_mbed.cpp

```diff
@@ -131,7 +131,9 @@ static void equeue_sema_timeout(equeue_sema_t *s) {
 bool equeue_sema_wait(equeue_sema_t *s, int ms) {
     int signal = 0;
     Timeout timeout;
-    if (ms > 0) {
+    if (ms == 0) {
+        return false;
+    } else if (ms >= 0) {
```

Comment:
Not a blocker for the PR, but it strictly greater than 0 might be clearer.
i also wonder what should happen when ms is less than 0 ? 
Is it expected that the semaphore will put the system to sleep without timeout ?